nixos/modules/services/network-filesystems/openafs/server.nix

```diff
@@ -101,6 +167,44 @@ in {
         description = "Definition of all cell-local database server machines.";
       };
 
+      authentication = {
+        principal =
+           let
+             defaultRealmPath = "security.krb5.settings.libdefaults.default_realm";
+             defaultRealm = config."${defaultRealmPath}";
+           in mkOption {
+             default = "afs/${defaultRealm}@${lib.toUpper defaultRealm}";
+             defaultText =
+               literalExpression "afs/\${config.${defaultRealmPath}}>@\${toUpper config.${defaultRealmPath}}";
```

Comment:
The cell name must still come from `cellName` and not assume `defaultRealmPath`. Also, `defaultRealmPath` is already in uppercase letters if it needs to be. Upcasing it unconditionally is wrong, because there might be instances where the Kerberos realm does not consist of all-upcase letters.

Reply:
Sorry, you are very much right, I didn't think it through properly...